pkg/agent/datamodel/types.go

```diff
@@ -512,6 +512,58 @@ type ProvisioningState string
 // AgentPoolProfileRole represents an agent role
 type AgentPoolProfileRole string
 
+// CustomKubeletConfig represents custom kubelet configurations for agent pool nodes
+type CustomKubeletConfig struct {
+	CPUManagerPolicy      string    `json:"cpuManagerPolicy,omitempty"`
```

Comment:
do the string fields not need to be pointer types to differentiate between user inputing an empty string vs not defined?

Reply:
Since we do not allow empty string for these fields, they do not have to be pointers. Actually making it a value type here blocks users from inputing empty strings and saving to DB, which I think is good.

Reply:
so an empty string here essentially means erase the previous values and do not override the setting, right?